File: backend/src/main/java/com/examreg/examreg/models/ExamRegistration.java
package com.examreg.examreg.models;

import java.time.LocalDateTime;

import com.fasterxml.jackson.annotation.JsonFormat;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@Entity
@AllArgsConstructor
@NoArgsConstructor
public class ExamRegistration {

  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  private Long id;

  @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
  private LocalDateTime registeredAt;

  @ManyToOne
  @JoinColumn(name = "student_id")
  private Student student;

  @ManyToOne
  @JoinColumn(name = "exam_session_id")
  private ExamSession examSession;
}

File: frontend/src/models/ExamRegistration.js
import { ExamSessionForStudent } from "./ExamSessionForStudent";

export class ExamRegistration {
  constructor(data) {
    this.id = data.id;
    this.studentId = data.studentId;
    this.registeredAt = data.registeredAt;
    this.examSession = data.examSession ? ExamSessionForStudent.fromJSON(data.examSession) : null;
  }

  static fromJSON(json) {
    return new ExamRegistration(json);
  }
}